akka-serialization-jackson/src/main/scala/akka/serialization/jackson/internal/AkkaJacksonSerializationFeature.scala

```diff
@@ -7,53 +7,225 @@ package akka.serialization.jackson.internal
+      RuntimeReflection.registerAllDeclaredMethods(clazz)
+
+      // FIXME we could do this only if Scala class to bloat Java apps less
+      registerClassResourceForAccess(access, clazz)
+
+      try {
+        val scalaModuleField = clazz.getDeclaredField("MODULE$")
+        RuntimeReflection.register(scalaModuleField)
+        log("Registering Scala module for " + clazz.getName)
+      } catch {
+        case _: NoSuchFieldException => // no scala module, this is ok
+      }
+
+      // getters
+      val scalaCaseClass = try {
+        clazz.getMethod("productElementNames")
+        true
+      } catch {
+        case _: NoSuchMethodException => false
+      }
+      util.Arrays
+        .stream(clazz.getDeclaredMethods)
+        .forEach(method =>
+          if (method.getParameterTypes.length == 0 && isPossiblyGetter(scalaCaseClass, method.getName)) {
+            log("Registering method " + clazz.getName + "." + method.getName)
+            RuntimeReflection.register(method)
+            RuntimeReflection.registerAsQueried(method)
+          })
+
+      RuntimeReflection.registerAllConstructors(clazz)
+      RuntimeReflection.registerAllDeclaredConstructors(clazz)
+      // we still need explicit register of each constructor for some reason
+      util.Arrays.stream(clazz.getDeclaredConstructors).forEach { constructor =>
+        log(
+          "Registering constructor " + clazz.getName + "(" + util.Arrays
+            .stream(constructor.getParameters)
+            .map(param => param.getName + ": " + param.getType.getName)
+            .collect(Collectors.joining(", ")) + ")")
+        RuntimeReflection.register(constructor)
+        RuntimeReflection.registerAsQueried(constructor)
+        RuntimeReflection.registerConstructorLookup(clazz, constructor.getParameterTypes: _*)
+        // also register each constructor parameter type
+        util.Arrays.stream(constructor.getParameterTypes).forEach { parameterType =>
+          if (parameterType.getPackage != null) {
+            val packageName = parameterType.getPackage.getName
+
+            // avoid digging to deep into akka and stdlib types
+            if (!packageName.startsWith("akka") && !packageName.startsWith("java") && !packageName
+                  .startsWith("scala")) {
+              registerTypeForJacksonSerialization(access, parameterType)
+
+              if (parameterType.isInterface) {
+                // ADT or something like it, try to register concrete classes for that interface as well
+                access.registerSubtypeReachabilityHandler({ (access, clazz) =>
+                  registerTypeForJacksonSerialization(access, clazz)
+                }, parameterType)
+              }
+              if (classOf[scala.Enumeration#Value].isAssignableFrom(parameterType)) {
+                warning(
+                  "Saw a scala.Enumeration field this is not supported out of the box and will require adding manual metadata")
```

Comment:
Some form of punctuation here:

```suggestion
                  "Saw a scala.Enumeration field. This is not supported out of the box and will require adding manual metadata")
```